cpp/src/io/comp/gpuinflate.cu

```diff
@@ -1220,5 +1221,27 @@ void gpu_copy_uncompressed_blocks(device_span<device_span<uint8_t const> const>
   }
 }
 
+__global__ void decompress_check_kernel(device_span<decompress_status const> stats,
+                                        bool* any_block_failure)
+{
+  auto tid = blockIdx.x * blockDim.x + threadIdx.x;
+  if (tid < stats.size()) {
+    if (stats[tid].status != 0) {
+      *any_block_failure = true;  // Doesn't need to be atomic
```

Comment:
Yes it does. 

https://eel.is/c++draft/basic#intro.races-21

Can't this just be a `thrust::reduce` over the `stats`?

Reply:
```
auto num_failed_blocks = thrust::count_if(stats.begin(), stats.end(),
 [] __device__ (auto s){
   return s.status != 0;
});